### PR TITLE
arduino-language-server: init at 0.7.1

### PR DESCRIPTION
**Pull request by BattleCh1cken**
###### Description of changes
Packaged the arduino-language server

https://github.com/arduino/arduino-language-server

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub }:
+
+{
+  pkg = buildGoModule rec {
```
Comment (AndersonTorres) [inline pkgs/development/embedded/arduino/arduino-language-server/default.nix:4]: have you ever read the manuals?

Comment (BattleCh1cken) [reply pkgs/development/embedded/arduino/arduino-language-server/default.nix:4]: Which one?

I'm assuming I did something wrong here but I have no idea what.

Comment (AndersonTorres) [inline pkgs/development/embedded/arduino/arduino-language-server/default.nix:4]:
```suggestion
  buildGoModule rec {
```

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    ldflags = [
+      "-s"
+      "-w"
+      "-X github.com/arduino/arduino-language-server/version.versionString=${version}"
+      "-X github.com/arduino/arduino-language-server/version.commit=unknown"
+    ] ++ lib.optionals stdenv.isLinux [ "-extldflags '-static'" ];
+
+    meta = with lib; {
+      inherit (src.meta) homepage;
+      description = "Arduino from the command line";
```
Comment (AndersonTorres) [inline pkgs/development/embedded/arduino/arduino-language-server/default.nix:30]: This is not an accurate description of a language server.

Indeed the page provides a better description.

```suggestion
      description = "An Arduino Language Server based on Clangd to Arduino code autocompletion";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix (hunk 2 above)
Comment (BattleCh1cken) [reply pkgs/development/embedded/arduino/arduino-language-server/default.nix:29]: Oops. I reused the expression from the cli. Sorry for the oversight. I'll make sure the rest of the metadata makes sense as well.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -1541,6 +1541,12 @@
     githubId = 576355;
     name = "Bas van Dijk";
   };
+  BattleCh1cken = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:1544]: Put this specific change in a separate commit. 
No need to open a pull-request.

Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:1544]: The other changes can be squashed in a single commit.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (BattleCh1cken) [reply maintainers/maintainer-list.nix:1544]: Alright. Thanks for helping me!

Comment (BattleCh1cken) [reply maintainers/maintainer-list.nix:1544]: Alrighty. All done.

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    "-X github.com/arduino/arduino-language-server/version.commit=unknown"
+  ] ++ lib.optionals stdenv.isLinux [ "-extldflags '-static'" ];
+
+  meta = with lib; {
+    inherit (src.meta) homepage;
+    description = "An Arduino Language Server based on Clangd to Arduino code autocompletion";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ BattleCh1cken ];
+  };
+
```
Comment (AndersonTorres) [inline pkgs/development/embedded/arduino/arduino-language-server/default.nix:32]:
```suggestion
```

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
```
Comment (AndersonTorres) [inline pkgs/development/embedded/arduino/arduino-language-server/default.nix:2]:
```suggestion

buildGoModule rec {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Also, squash all the commits (except the `maintainers` file) into one.

**Comment by BattleCh1cken** [comment]
Alright, my changes seem to actually have applied this time. When I pulled after squashing it changed it back to what it was. This time it's correct.

**Comment by AndersonTorres** [comment]
Now remove all the useless text you left on the second commit, leaving only the title `arduino-language-server: init at 0.7.1`

Also do not use ending period on the commit title.

**Comment by AndersonTorres** [comment]
Welcome aboard!
